dotnet/samples/Concepts/Memory/VectorstoreLangchainInterop/AzureAISearchFactory.cs

```diff
@@ -0,0 +1,137 @@
+            }
+
+            // Create an Azure AI Search collection. To be compatible with Langchain
+            // we need to use a custom record definition that matches the
+            // schema used by Lanchain. We also need to use a custom mapper
+            // since the Langchain schema includes a metadata field that is
+            // a JSON string containing the source property. Parsing this
+            // string and extracting the source is not supported by the default mapper.
+            return (new AzureAISearchVectorStoreRecordCollection<TRecord>(
+                searchIndexClient,
+                name,
+                new()
+                {
+                    VectorStoreRecordDefinition = s_recordDefinition,
+                    JsonObjectCustomMapper = new LangchainInteropMapper() as IVectorStoreRecordMapper<TRecord, JsonObject>
+                }) as IVectorStoreRecordCollection<TKey, TRecord>)!;
+        }
+    }
+
+    /// <summary>
+    /// Custom mapper to map the metadata string field, since it contains JSON as a string and this is not supported
+    /// automatically by the built in mapper.
+    /// </summary>
+    private sealed class LangchainInteropMapper : IVectorStoreRecordMapper<LangchainDocument<string>, JsonObject>
+    {
+        public JsonObject MapFromDataToStorageModel(LangchainDocument<string> dataModel)
+        {
+            var storageDocument = new AzureAISearchLangchainDocument()
+            {
+                Key = dataModel.Key,
+                Content = dataModel.Content,
+                Metadata = $"{{\"source\": \"{dataModel.Source}\"}}",
+                Embedding = dataModel.Embedding
+            };
+
+            return JsonSerializer.SerializeToNode(storageDocument)!.AsObject();
+        }
+
+        public LangchainDocument<string> MapFromStorageToDataModel(JsonObject storageModel, StorageToDataModelMapperOptions options)
+        {
+            var storageDocument = JsonSerializer.Deserialize<AzureAISearchLangchainDocument>(storageModel)!;
+            var metadataDocument = JsonSerializer.Deserialize<JsonObject>(storageDocument.Metadata);
+            var source = metadataDocument?["source"]?.AsValue()?.ToString();
+
+            return new LangchainDocument<string>()
+            {
+                Key = storageDocument.Key,
+                Content = storageDocument.Content,
+                Source = source!,
+                Embedding = storageDocument.Embedding
+            };
+        }
+    }
+
+    /// <summary>
+    /// Model class that matches the storage format used by Langchain for Azure AI Search.
+    /// </summary>
+    private sealed class AzureAISearchLangchainDocument
+    {
+        [JsonPropertyName("id")]
+        public string Key { get; set; }
+
+        [JsonPropertyName("content")]
+        public string Content { get; set; }
+
+        /// <summary>
+        /// The storage format used by Lanchain stores the source information
```

Comment:
nit: A tiny typo - Lan**g**chain here and in a few other places

Reply:
Thanks fixed.